src/rust/Cargo.toml

```diff
@@ -48,3 +48,5 @@ lto = true
 
 [profile.dev]
 opt-level = 'z'
+# Right now lto="thin" takes to much space, we have to have "fat"/true
+lto = true
```

Comment:
Maybe `codegen-units = 1` also here would help with the size?

`debug = true` in the release build but not the dev build seems counter-intuitive.

Reply:
`debug = true` is the default in debug builds. This is simply to add elf debug info to the elf binary. That debug info is not copied when creating the binary file we eventually flash.